### ISSUE TITLE
Exception while resharding of keyspace: Getting table state is allowed only with static sharding (tablets are enabled)

### ISSUE THREAD

**Issue by juliayakovlev** [scylladb/scylladb#20912]
## Packages

Scylla version: `6.2.0~rc0-20240917.cbc53f0e81f5` with build-id `a76b6bece2744410fa062558bb8df6a360556565`
Target Scylla version: `6.3.0~dev-0.20240925.ae764814441a` with build-id `69dba99aba3c8d894ee027a5708139963442c206`
Kernel Version: `6.8.0-1016-aws`

## Issue description

- [ ] This issue is a regression.
- [ ] It is unknown if this issue is a regression.

Tablets are enabled.
the issue does not happen when tablets are disabled.
Upgrade from `6.2.0~rc0-20240917.cbc53f0e81f5`  to `6.3.0~dev-0.20240925.ae764814441a`.
Scylla failed to start up **after** upgrade.
Exception during resharding `keyspace1.standard1` table (while the load running) after upgrade:

```
Sep 26 10:17:54.312093 perf-latency-upgrade-ubuntu-db-node-71e95ef6-1 scylla[15704]:  [shard 0:main] database - Keyspace keyspace1: Reading CF standard1 id=521e4590-7bd9-11ef-a2c4-690971ba285f version=521e4591-7bd9-11ef-a2c4-690971ba285f storage=LOCAL datadir=/var/lib/scylla/data/keyspace1/standard1-521e45907bd911efa2c4690971ba285f
Sep 26 10:17:55.536345 perf-latency-upgrade-ubuntu-db-node-71e95ef6-1 scylla[15704]:  [shard 0:main] database - Resharding 6GB for keyspace1.standard1
Sep 26 10:17:55.537324 perf-latency-upgrade-ubuntu-db-node-71e95ef6-1 scylla[15704]:  [shard 0:main] database - Exception while populating keyspace 'keyspace1' with column family 'standard1' from datadir '/var/lib/scylla/data/keyspace1/standard1-521e45907bd911efa2c4690971ba285f': std::runtime_error (Getting table state is allowed only with static sharding)
Sep 26 10:17:55.539649 perf-latency-upgrade-ubuntu-db-node-71e95ef6-1 scylla[15704]:  [shard 0:main] init - Shutting down Raft API
Sep 26 10:17:55.540118 perf-latency-upgrade-ubuntu-db-node-71e95ef6-1 scylla[15704]:  [shard 0:main] init - Shutting down Raft API was successful
Sep 26 10:17:55.540124 perf-latency-upgrade-ubuntu-db-node-71e95ef6-1 scylla[15704]:  [shard 0:main] init - Shutting down sstables format listener
Sep 26 10:17:55.540127 perf-latency-upgrade-ubuntu-db-node-71e95ef6-1 scylla[15704]:  [shard 0:main] init - Shutting down sstables format listener was successful
Sep 26 10:17:55.540131 perf-latency-upgrade-ubuntu-db-node-71e95ef6-1 scylla[15704]:  [shard 0:main] init - Shutting down query processor remote part
Sep 26 10:17:55.540424 perf-latency-upgrade-ubuntu-db-node-71e95ef6-1 scylla[15704]:  [shard 0:main] init - Shutting down query processor remote part was successful
Sep 26 10:17:55.540432 perf-latency-upgrade-ubuntu-db-node-71e95ef6-1 scylla[15704]:  [shard 0:main] init - Shutting down storage service API
Sep 26 10:17:55.540919 perf-latency-upgrade-ubuntu-db-node-71e95ef6-1 scylla[15704]:  [shard 0:main] init - Shutting down storage service API was successful
Sep 26 10:17:55.540926 perf-latency-upgrade-ubuntu-db-node-71e95ef6-1 scylla[15704]:  [shard 0:main] init - Shutting down storage_service

```

## Impact

Scylla can not start

## How frequently does it reproduce?

It happens every upgrade test with tablets enabled. 
I did not find opened issue

## Installation details

Cluster size: 6 nodes (i3en.2xlarge)

Scylla Nodes used in this run:
- perf-latency-upgrade-ubuntu-db-node-71e95ef6-6 (3.255.114.250 | 10.4.7.123) (shards: 7)
- perf-latency-upgrade-ubuntu-db-node-71e95ef6-5 (18.201.202.66 | 10.4.5.58) (shards: 7)
- perf-latency-upgrade-ubuntu-db-node-71e95ef6-4 (54.228.176.188 | 10.4.7.141) (shards: 7)
- perf-latency-upgrade-ubuntu-db-node-71e95ef6-3 (34.240.223.223 | 10.4.5.214) (shards: 7)
- perf-latency-upgrade-ubuntu-db-node-71e95ef6-2 (18.201.237.195 | 10.4.6.6) (shards: 7)
- perf-latency-upgrade-ubuntu-db-node-71e95ef6-1 (34.253.68.14 | 10.4.7.161) (shards: -1)

OS / Image: `ami-0af227e94a100d678` (aws: undefined_region)

Test: `scylla-master-perf-regression-latency-650gb-during-rolling-upgrade-tablets`
Test id: `71e95ef6-26f7-44ec-ad4c-6359e065ab8b`
Test name: `scylla-master/perf-regression/scylla-master-perf-regression-latency-650gb-during-rolling-upgrade-tablets`
Test method: `performance_regression_test.PerformanceRegressionUpgradeTest.test_latency_write_with_upgrade`
Test config file(s):

- [perf-regression-latency-650gb-upgrade.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/0abca81bdcc589bb259b6d92e34277f615e4f414/test-cases/performance/perf-regression-latency-650gb-upgrade.yaml)

<details>
<summary>
Logs and commands
</summary>

- Restore Monitor Stack command: `$ hydra investigate show-monitor 71e95ef6-26f7-44ec-ad4c-6359e065ab8b`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=71e95ef6-26f7-44ec-ad4c-6359e065ab8b)
- Show all stored logs command: `$ hydra investigate show-logs 71e95ef6-26f7-44ec-ad4c-6359e065ab8b`

## Logs:
- **db-cluster-71e95ef6.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/71e95ef6-26f7-44ec-ad4c-6359e065ab8b/20240926_102228/db-cluster-71e95ef6.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/71e95ef6-26f7-44ec-ad4c-6359e065ab8b/20240926_102228/db-cluster-71e95ef6.tar.gz)
- **sct-runner-events-71e95ef6.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/71e95ef6-26f7-44ec-ad4c-6359e065ab8b/20240926_102228/sct-runner-events-71e95ef6.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/71e95ef6-26f7-44ec-ad4c-6359e065ab8b/20240926_102228/sct-runner-events-71e95ef6.tar.gz)
- **sct-71e95ef6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/71e95ef6-26f7-44ec-ad4c-6359e065ab8b/20240926_102228/sct-71e95ef6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/71e95ef6-26f7-44ec-ad4c-6359e065ab8b/20240926_102228/sct-71e95ef6.log.tar.gz)
- **loader-set-71e95ef6.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/71e95ef6-26f7-44ec-ad4c-6359e065ab8b/20240926_102228/loader-set-71e95ef6.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/71e95ef6-26f7-44ec-ad4c-6359e065ab8b/20240926_102228/loader-set-71e95ef6.tar.gz)
- **monitor-set-71e95ef6.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/71e95ef6-26f7-44ec-ad4c-6359e065ab8b/20240926_102228/monitor-set-71e95ef6.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/71e95ef6-26f7-44ec-ad4c-6359e065ab8b/20240926_102228/monitor-set-71e95ef6.tar.gz)

[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/perf-regression/job/scylla-master-perf-regression-latency-650gb-during-rolling-upgrade-tablets/6/)
[Argus](https://argus.scylladb.com/test/fd24cecf-b43e-4230-831c-5c80a3ee04b7/runs?additionalRuns[]=71e95ef6-26f7-44ec-ad4c-6359e065ab8b)
</details>

**Comment by denesb** [comment]
We should not reshard at all when tablets are enabled. It is not supported.

**Comment by denesb** [comment]
It is not harmless. We attempt to reshard a tablets table, which results in basically UB, as we are executing code-paths that are not tablet aware (as they don't have to be). We added checks against this: https://github.com/scylladb/scylladb/pull/17008. I wonder why it failed here.

**Comment by fruch** [comment]
seems like version 
`6.3.0~dev-20240927.c17d35371846`

was working o.k. in this week rolling upgrade test:
https://argus.scylladb.com/test/cc80460f-2278-4fe5-8781-1776502bc7e1/runs?additionalRuns[]=8caddf9c-7bdc-4e1b-b2d5-decf4acc74e3

triggered it again, to see if this reproduce also in the master rolling upgrade tests.

**Comment by juliayakovlev** [comment]
All performance upgrade tests with tablets failed due to this issue

https://argus.scylladb.com/tests/scylla-cluster-tests/0f202bc5-f8b8-4a22-b2fa-f1857f86a6b2

**Comment by denesb** [comment]
Evidently `collect_all_shared_sstables()` finds sstables that it thinks either need cleanup and/or reshard. I'll try to reproduce locally.

**Comment by denesb** [comment]
I failed to reproduce locally.

I am looking at the list of commits between master and  6.3.0~dev-20240927.c17d35371846 (the last known working version).
I think there is a reasonable chance that https://github.com/scylladb/scylladb/pull/20819 is the culprit, although I don't have any proof (either pro or con).

**Comment by denesb** [comment]
There are two ways `collect_all_shared_sstables()` can decide an sstables needs to be resharded:
* if `needs_cleanup()` returns `true`
* if `database_directory` placed the sstable into `_shared_sstables`

When populating the table directory, `reshard()` is called with `owned_ranges_ptr = nullptr`, so the first check is short-circuited (to `false`). So we are dealing with the second case here, somehow `sstable_directory` considers these sstables shared.

**Comment by denesb** [comment]
@juliayakovlev can you re-run the test and keep at least one node after, so I can have a look at it?

**Comment by denesb** [comment]
> > [@juliayakovlev](https://github.com/juliayakovlev) can you re-run the test and keep at least one node after, so I can have a look at it?
> 
> [@denesb](https://github.com/denesb) the issue is reproduced [here](https://argus.scylladb.com/tests/scylla-cluster-tests/d2e0d64f-2d44-4cb1-ac3b-e966eb12adf2) with keep cluster. You can find nodes list in the "Resources" tab

Thanks, I will have a look.

**Comment by denesb** [comment]
Looks like we find an sstable we consider shared:
```
TRACE 2024-11-07 13:36:10,794 [shard 0:main] sstable_directory - /var/lib/scylla/data/keyspace1/standard1-1ab6c1909c3111ef81a428ec30a12787/me-3gl0_11ok_3elkw2lrcsgtbyo0if-big-Data.db identified as a shared SSTable, shards=[6, 3]
```

No sharding metadata in scylla-metadata:
```
$ ./scylladb/bin/scylla sstable dump-scylla-metadata /var/lib/scylla/data/keyspace1/standard1-1ab6c1909c3111ef81a428ec30a12787/me-3gl0_11ok_3elkw2lrcsgtbyo0if-big-Data.db
{"sstables":{"/var/lib/scylla/data/keyspace1/standard1-1ab6c1909c3111ef81a428ec30a12787/me-3gl0_11ok_3elkw2lrcsgtbyo0if-big-Data.db":{"sharding":[],"features":{"mask":63,"features":["NonCompoundRangeTombstones","ShadowableTombstones","CorrectStaticCompact","CorrectEmptyCounters","CorrectUDTsInCollections","CorrectLastPiBlockWidth"]},"run_identifier":"c0930a14-7468-4458-ba59-cd309df1596f","large_data_stats":{"partition_size":{"max_value":1117,"threshold":1048576000,"above_threshold":0},"rows_in_partition":{"max_value":1,"threshold":100000,"above_threshold":0},"row_size":{"max_value":1092,"threshold":10485760,"above_threshold":0},"cell_size":{"max_value":136,"threshold":1048576,"above_threshold":0},"elements_in_collection":{"max_value":0,"threshold":10000,"above_threshold":0}},"sstable_origin":"split","scylla_version":"6.2.0-0.20241013.b8a9fd4e49e8","scylla_build_id":"a61f658b0408ba10663812f7a3b4d6aea7714fac","ext_timestamp_stats":{"min_live_timestamp":1730892028966003,"min_live_row_marker_timestamp":9223372036854775807}}}}
```

I managed to reproduce locally on my machine with the downloaded sstable. I will continue the investigation tomorrow.

**Comment by denesb** [comment]
It appears that the sstable genuinely spans 2 tablets. It could be a problem with a previous split.

**Comment by denesb** [comment]
I see a lot of these in the logs:
```
Nov 01 00:41:59.001360 perf-latency-upgrade-ubuntu-db-node-0f202bc5-1 scylla[6357]:  [shard 1:strm] compaction - [Split keyspace1.standard1 1a3b2580-97ea-11ef-83b9-6c0ea844b4d7] Splitting of 1 sstables interrupted due to: sstables::compaction_stopped_exception (Compaction for keyspace1/standard1 was stopped due to: user-triggered operation), at 0x6169f0e 0x616a520 0x616a828 0x253fbaa 0x253df4e 0x60406c6                                                                                                        
                                                                                       --------                                                                                                                                                                
                                                                                       seastar::continuation<seastar::internal::promise_base_with_type<sstables::compaction_result>, seastar::async<sstables::compaction::run(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >)::$_0>(seastar::thread_attributes, sstables::compaction::run(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >)::$_0&&)::{lambda()#2}, seastar::future<void>::then_impl_nrvo<seastar::async<sstables::compaction::run(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >)::$_0>(seastar::thread_attributes, sstables::compaction::run(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >)::$_0&&)::{lambda()#2}, seastar::future<sstables::compaction_result> >(sstables::compaction::run(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >)::$_0&&)::{lambda(seastar::internal::promise_base_with_type<sstables::compaction_result>&&, seastar::async<sstables::compaction::run(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >)::$_0>(seastar::thread_attributes, auto:1&&, (auto:2&&)...)::{lambda()#2}&, seastar::future_state<seastar::internal::monostate>&&)#1}, void>                                                                                                                                                                                                                             
                                                                                       --------                                                                                                                                                                
                                                                                       seastar::continuation<seastar::internal::promise_base_with_type<sstables::compaction_result>, seastar::future<sstables::compaction_result>::finally_body<seastar::async<sstables::compaction::run(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >)::$_0>(seastar::thread_attributes, sstables::compaction::run(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >)::$_0&&)::{lambda()#3}, false>, seastar::future<sstables::compaction_result>::then_wrapped_nrvo<seastar::future<sstables::compaction_result>, seastar::future<sstables::compaction_result>::finally_body<seastar::async<sstables::compaction::run(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >)::$_0>(seastar::thread_attributes, sstables::compaction::run(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >)::$_0&&)::{lambda()#3}, false> >(seastar::future<sstables::compaction_result>::finally_body<seastar::async<sstables::compaction::run(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >)::$_0>(seastar::thread_attributes, sstables::compaction::run(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >)::$_0&&)::{lambda()#3}, false>&&)::{lambda(seastar::internal::promise_base_with_type<sstables::compaction_result>&&, seastar::future<sstables::compaction_result>::finally_body<seastar::async<sstables::compaction::run(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >)::$_0>(seastar::thread_attributes, auto:1&&, (auto:2&&)...)::{lambda()#3}, false>&, seastar::future_state<sstables::compaction_result>&&)#1}, sstables::compaction_result>                                                                                                                                                                                                                                                
                                                                                       --------                                                                                                                                                                
                                                                                       seastar::internal::coroutine_traits_base<void>::promise_type                                                                                                            
                                                                                       --------                                                                                                                                                                
                                                                                       seastar::internal::coroutine_traits_base<std::optional<sstables::compaction_stats> >::promise_type                                                                      
                                                                                       --------                                                                                                                                                                
                                                                                       seastar::shared_future<std::optional<sstables::compaction_stats> >::shared_state                                                                                        
Nov 01 00:41:59.001415 perf-latency-upgrade-ubuntu-db-node-0f202bc5-1 scylla[6357]:  [shard 1:strm] compaction_manager - Split SSTable task 0x601004d5c400 for table keyspace1.standard1 compaction_group=31 [0x601004b6ab20]: stopped, reason: Compaction for keyspace1/standard1 was stopped due to: user-triggered operation 
```
But I couldn't yet figure out why we are stopping random slit compactions. It is possible that the bug is that we don't retry some of these and some sstables remain unsplit.

**Comment by denesb** [comment]
There are three operations which stop all compaction with this message: cleanup, rewrite and split. Could we have two splits racing here?

**Comment by raphaelsc** [comment]
I think the root cause is likely https://github.com/scylladb/scylladb/issues/20626, since 6.2.0-20241013.b8a9fd4e49e8 is affected by it. I'd suggest retrying on some later version, upgrading from some version containing 38ce2c605dc97d0ed13a7fc241e0 to some later version. We tried to reshard, because we're trying to load a sstable which we failed to load into the sstable set due to the bug, and it can happen that range spanned by it was split, and one half was split into a different shard of same node or another, therefore sharder returns that sstable belongs to more than one shard. As for the reason we're stopping some split compaciton, I think it's unrelated, it happens due to migrations happening in parallel to split, and migration has higher priority so it stops all compaction work (including split) and continue on the destination.

**Comment by raphaelsc** [comment]
this is evidence for it:

```
$ for i in perf-latency-upgrade-ubuntu-db-node-d2e0d64f-*/messages.log; do echo $i; cat $i | grep "Failed to load"; done
perf-latency-upgrade-ubuntu-db-node-d2e0d64f-1/messages.log
2024-11-06T13:47:18.039+00:00 perf-latency-upgrade-ubuntu-db-node-d2e0d64f-1      !ERR | scylla[6358]:  [shard 3:strm] table - Failed to load SSTable /var/lib/scylla/data/keyspace1/standard1-1ab6c1909c3111ef81a428ec30a12787/me-3gl0_11ok_3elkw2lrcsgtbyo0if-big-TOC.txt: sstables::compaction_stopped_exception (Compaction for keyspace1/standard1 was stopped due to: user-triggered operation)
perf-latency-upgrade-ubuntu-db-node-d2e0d64f-2/messages.log
2024-11-06T13:26:28.794+00:00 perf-latency-upgrade-ubuntu-db-node-d2e0d64f-2      !ERR | scylla[6347]:  [shard 1:strm] table - Failed to load SSTable /var/lib/scylla/data/keyspace1/standard1-1ab6c1909c3111ef81a428ec30a12787/me-3gl0_10pr_2fg5c2fz800xx0cpq5-big-TOC.txt: sstables::compaction_stopped_exception (Compaction for keyspace1/standard1 was stopped due to: user-triggered operation)
2024-11-06T13:47:42.044+00:00 perf-latency-upgrade-ubuntu-db-node-d2e0d64f-2      !ERR | scylla[6347]:  [shard 3:strm] table - Failed to load SSTable /var/lib/scylla/data/keyspace1/standard1-1ab6c1909c3111ef81a428ec30a12787/me-3gl0_11ok_3sbeo2fz800xx0cpq5-big-TOC.txt: sstables::compaction_stopped_exception (Compaction for keyspace1/standard1 was stopped due to: user-triggered operation)
perf-latency-upgrade-ubuntu-db-node-d2e0d64f-3/messages.log
2024-11-06T13:27:51.770+00:00 perf-latency-upgrade-ubuntu-db-node-d2e0d64f-3      !ERR | scylla[6350]:  [shard 1:strm] table - Failed to load SSTable /var/lib/scylla/data/keyspace1/standard1-1ab6c1909c3111ef81a428ec30a12787/me-3gl0_10pm_501hc2n323azb3c3i9-big-TOC.txt: sstables::compaction_stopped_exception (Compaction for keyspace1/standard1 was stopped due to: user-triggered operation)
2024-11-06T13:47:10.541+00:00 perf-latency-upgrade-ubuntu-db-node-d2e0d64f-3      !ERR | scylla[6350]:  [shard 3:strm] table - Failed to load SSTable /var/lib/scylla/data/keyspace1/standard1-1ab6c1909c3111ef81a428ec30a12787/me-3gl0_11ok_3qdyo2n323azb3c3i9-big-TOC.txt: sstables::compaction_stopped_exception (Compaction for keyspace1/standard1 was stopped due to: user-triggered operation)
2024-11-06T13:47:10.541+00:00 perf-latency-upgrade-ubuntu-db-node-d2e0d64f-3      !ERR | scylla[6350]:  [shard 2:strm] table - Failed to load SSTable /var/lib/scylla/data/keyspace1/standard1-1ab6c1909c3111ef81a428ec30a12787/me-3gl0_11ok_3q68w2e0ejvsh3w8u9-big-TOC.txt: sstables::compaction_stopped_exception (Compaction for keyspace1/standard1 was stopped due to: user-triggered operation)
perf-latency-upgrade-ubuntu-db-node-d2e0d64f-4/messages.log
2024-11-06T13:26:10.774+00:00 perf-latency-upgrade-ubuntu-db-node-d2e0d64f-4      !ERR | scylla[6338]:  [shard 1:strm] table - Failed to load SSTable /var/lib/scylla/data/keyspace1/standard1-1ab6c1909c3111ef81a428ec30a12787/me-3gl0_10pm_5oots2f725m7qkmsah-big-TOC.txt: sstables::compaction_stopped_exception (Compaction for keyspace1/standard1 was stopped due to: user-triggered operation)
2024-11-06T13:47:01.274+00:00 perf-latency-upgrade-ubuntu-db-node-d2e0d64f-4      !ERR | scylla[6338]:  [shard 2:strm] table - Failed to load SSTable /var/lib/scylla/data/keyspace1/standard1-1ab6c1909c3111ef81a428ec30a12787/me-3gl0_11ok_3qte825qmijncntkmx-big-TOC.txt: sstables::compaction_stopped_exception (Compaction for keyspace1/standard1 was stopped due to: user-triggered operation)
perf-latency-upgrade-ubuntu-db-node-d2e0d64f-5/messages.log
2024-11-06T13:27:01.833+00:00 perf-latency-upgrade-ubuntu-db-node-d2e0d64f-5      !ERR | scylla[6362]:  [shard 1:strm] table - Failed to load SSTable /var/lib/scylla/data/keyspace1/standard1-1ab6c1909c3111ef81a428ec30a12787/me-3gl0_10pr_2uvpc1yz1hrxifekqv-big-TOC.txt: sstables::compaction_stopped_exception (Compaction for keyspace1/standard1 was stopped due to: user-triggered operation)
2024-11-06T13:47:24.333+00:00 perf-latency-upgrade-ubuntu-db-node-d2e0d64f-5      !ERR | scylla[6362]:  [shard 3:strm] table - Failed to load SSTable /var/lib/scylla/data/keyspace1/standard1-1ab6c1909c3111ef81a428ec30a12787/me-3gl0_11ok_3ovy81yz1hrxifekqv-big-TOC.txt: sstables::compaction_stopped_exception (Compaction for keyspace1/standard1 was stopped due to: user-triggered operation)
2024-11-06T13:48:00.583+00:00 perf-latency-upgrade-ubuntu-db-node-d2e0d64f-5      !ERR | scylla[6362]:  [shard 2:strm] table - Failed to load SSTable /var/lib/scylla/data/keyspace1/standard1-1ab6c1909c3111ef81a428ec30a12787/me-3gl0_11ok_3pbds26ckg7bwsucnb-big-TOC.txt: sstables::compaction_stopped_exception (Compaction for keyspace1/standard1 was stopped due to: user-triggered operation)
perf-latency-upgrade-ubuntu-db-node-d2e0d64f-6/messages.log
2024-11-06T13:25:46.534+00:00 perf-latency-upgrade-ubuntu-db-node-d2e0d64f-6      !ERR | scylla[6352]:  [shard 1:strm] table - Failed to load SSTable /var/lib/scylla/data/keyspace1/standard1-1ab6c1909c3111ef81a428ec30a12787/me-3gl0_10pr_4f1b42n36d9akgmyn2-big-TOC.txt: sstables::compaction_stopped_exception (Compaction for keyspace1/standard1 was stopped due to: user-triggered operation)
2024-11-06T13:44:19.284+00:00 perf-latency-upgrade-ubuntu-db-node-d2e0d64f-6      !ERR | scylla[6352]:  [shard 3:strm] table - Failed to load SSTable /var/lib/scylla/data/keyspace1/standard1-1ab6c1909c3111ef81a428ec30a12787/me-3gl0_11ok_3pbds2n36d9akgmyn2-big-TOC.txt: sstables::compaction_stopped_exception (Compaction for keyspace1/standard1 was stopped due to: user-triggered operation)
2024-11-06T13:46:18.034+00:00 perf-latency-upgrade-ubuntu-db-node-d2e0d64f-6      !ERR | scylla[6352]:  [shard 2:strm] table - Failed to load SSTable /var/lib/scylla/data/keyspace1/standard1-1ab6c1909c3111ef81a428ec30a12787/me-3gl0_11ok_3pbds2ry8sl3g8llni-big-TOC.txt: sstables::compaction_stopped_exception (Compaction for keyspace1/standard1 was stopped due to: user-triggered operation)
```

**Comment by denesb** [comment]
@juliayakovlev ping, can you repeat the test according to @raphaelsc's recommendation, so we can see if this is indeed an already fixed issue, or we need to dig deeper?

**Comment by juliayakovlev** [comment]
> [@juliayakovlev](https://github.com/juliayakovlev) ping, can you repeat the test according to [@raphaelsc](https://github.com/raphaelsc)'s recommendation, so we can see if this is indeed an already fixed issue, or we need to dig deeper?

@denesb 
I see that [this commit](https://github.com/scylladb/scylladb/commit/38ce2c605dc97d0ed13a7fc241e086c768e03d35) was backported to 6.2, 6.1 and 6.0. So can I run upgrade from 6.1 to 6.2?

**Comment by denesb** [comment]
> > [@juliayakovlev](https://github.com/juliayakovlev) ping, can you repeat the test according to [@raphaelsc](https://github.com/raphaelsc)'s recommendation, so we can see if this is indeed an already fixed issue, or we need to dig deeper?
> 
> [@denesb](https://github.com/denesb) I see that [this commit](https://github.com/scylladb/scylladb/commit/38ce2c605dc97d0ed13a7fc241e086c768e03d35) was backported to 6.2, 6.1 and 6.0. So can I run upgrade from 6.1 to 6.2?

Yes, just be sure to use the latest version which has the fix already.

**Comment by juliayakovlev** [comment]
@denesb @raphaelsc 

The [commit](https://github.com/scylladb/scylladb/commit/38ce2c605dc97d0ed13a7fc241e086c768e03d35) was backported:
- to `branch-6.1`  - Oct, 17
- to `branch-6.2` - 2 weeks ago

I ran upgrade test from `6.1.3-20241027.91c693bf9332` to `6.2.1-20241106.a3a0ffbcd015`.
https://argus.scylladb.com/tests/scylla-cluster-tests/731e0994-5e24-4bd5-85f9-b452dc4e0d53

The error described above is not reproduced

**Comment by raphaelsc** [comment]
Good, looks like we can close this one.

**Comment by denesb** [comment]
Closing as duplicate of https://github.com/scylladb/scylladb/issues/20626.
